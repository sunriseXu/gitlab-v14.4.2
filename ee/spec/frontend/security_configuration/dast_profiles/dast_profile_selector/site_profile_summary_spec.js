import { shallowMount } from '@vue/test-utils';
import App from 'ee/security_configuration/dast_profiles/dast_profile_selector/site_profile_summary.vue';
import { siteProfiles } from 'ee_jest/security_configuration/dast_profiles/mocks/mock_data';

const [profile] = siteProfiles;

describe('DastScannerProfileSummary', () => {
  let wrapper;

  const createWrapper = (props = {}) => {
    wrapper = shallowMount(App, {
      propsData: {
        profile,
        ...props,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders properly', () => {
    createWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });
});
