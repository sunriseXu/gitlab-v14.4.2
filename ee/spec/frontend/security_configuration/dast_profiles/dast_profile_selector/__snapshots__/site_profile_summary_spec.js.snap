// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DastScannerProfileSummary renders properly 1`] = `
<dast-profile-summary-card-stub
  iseditable="true"
  profiletype="site"
>
  
    Validation pending
  
  <summary-cell-stub
    class=""
    label="Target URL"
    value="http://pending.test"
  />
   
  <summary-cell-stub
    label="Site type"
    value="Website"
  />
   
  <summary-cell-stub
    label="Authentication URL"
    value="http://pending.test/sign-in"
  />
   
  <summary-cell-stub
    label="Username"
    value="validation-pending@example.com"
  />
   
  <summary-cell-stub
    label="Password"
    value="••••••••"
  />
   
  <summary-cell-stub
    label="Username form field"
    value="session[username]"
  />
   
  <summary-cell-stub
    label="Password form field"
    value="session[password]"
  />
   
  <summary-cell-stub
    label="Submit button"
  />
   
  <summary-cell-stub
    label="Excluded URLs"
    value="http://pending.test/sign-out,http://pending.test/hidden"
  />
   
  <summary-cell-stub
    label="Request headers"
  />
   
  <summary-cell-stub
    label="Validation status"
  >
    <dast-site-validation-badge-stub
      status="PENDING_VALIDATION"
    />
  </summary-cell-stub>
</dast-profile-summary-card-stub>
`;
